spec/imsc-hrm.html

```diff
@@ -823,14 +815,20 @@ <h2>Paint Text</h2>
       height)<sup>2</sup></p>
 
       <p class='note'>NRGA(G<sub>i</sub>) does not take into account decorations (e.g. underline), effects (e.g. outline) or actual
-      typographical glyph aspect ratio. An implementation can determine an actual buffer size needs based on worst-case glyph size
+      typographical glyph aspect ratio. An implementation can determine an actual cache size needs based on worst-case glyph size
       complexity.</p>
 
-      <p>The contents of the Glyph Buffer G<sub>n</sub> are copied instantaneously to Glyph Buffer G<sub>n-1</sub> at the
-      presentation time of <a>Intermediate Synchronic Document</a> E<sub>n</sub>.</p>
+      <p>At the presentation time of <a>Intermediate Synchronic Document</a> E<sub>n</sub>:</p>
+      <ol>
+        <li>purge from the Glyph Cache all <a>glyphs</a> not flagged <i>retain</i>; and</li>
+        <li>remove the <i>retain</i> flag from all remaining <a>glyphs</a> in the Glyph Cache; and</li>
+      </ol>
+
+      <p>It SHALL be an <a>error</a> for the sum of NRGA(g<sub>i</sub>) over all <a data-lt="glyph">glyphs</a> in the
```

Comment:
Same comment as for the image cache regarding when this error condition is checked, and whether glyphs for discard should be purged before drawing non-retained glyphs.

Reply:
Looking at [IMSC 1.2](https://www.w3.org/TR/ttml-imsc1.2/#paint-images), the condition was imposed on the Glyph Buffer G_n, which I think corresponds to all glyphs flagged _retain_.

https://github.com/w3c/imsc-hrm/pull/55/commits/838e92f60101b8ebc2c544602b60a6c903359b60

Reply:
Thanks, that works I think. Not sure why the phrasing is different for glyphs and images, but it's a v minor point ("if the sum is" vs "for the sum to be").

Reply:
No reason I can think of. Fixed.